domain/src/main/java/org/oppia/android/domain/topic/TopicController.kt

```diff
@@ -161,18 +161,20 @@ class TopicController @Inject constructor(
     topicId: String,
     storyId: String,
     explorationId: String
-  ): LiveData<AsyncResult<ChapterSummary>> {
-    return MutableLiveData(
-      try {
-        AsyncResult.success(fetchChapter(topicId, storyId, explorationId))
-      } catch (e: Exception) {
+  ): DataProvider<ChapterSummary> {
+    return dataProviders.createInMemoryDataProviderAsync(GET_STORY_PROVIDER_ID) {
+      return@createInMemoryDataProviderAsync AsyncResult.success(retrieveStory(topicId, storyId))
+    }.transformAsync(GET_CHAPTER_PROVIDER_ID) { storySummary ->
+      val chapterSummary = fetchChapter(storySummary, explorationId)
+      if (chapterSummary == null)
```

Comment:
This if/else needs braces per https://developer.android.com/kotlin/style-guide#braces.

Reply:
A nit: but I also suggest inverting the if/else condition so that the default cause is success (since that's the expected happy path).

Reply:
done